src/app/add-course-page/calendar/calendar.component.ts

```diff
@@ -1,21 +1,23 @@
-import { Component, Input, OnInit, Output, EventEmitter } from '@angular/core';
+import { Component, Input, Output, EventEmitter, DoCheck } from '@angular/core';
 
 @Component({
   selector: 'app-calendar',
   templateUrl: './calendar.component.html',
   styleUrls: ['./calendar.component.scss']
 })
-export class CalendarComponent implements OnInit {
+export class CalendarComponent implements DoCheck {
 
-  @Input() creationDate: Date;
+  @Input() creationDate: Date | string;
   @Output() creationDateChange: EventEmitter<Date> = new EventEmitter()
 
   placeholderData: string;
 
-  ngOnInit(): void {
-    if (this.creationDate) {
-      Number(this.creationDate) - Number(new Date()) > -100 && Number(this.creationDate) - Number(new Date()) < 0 ? this.placeholderData = 'Choose a date' :
-        this.placeholderData = `${this.creationDate.getDate()}/${this.creationDate.getMonth()}/${this.creationDate.getFullYear()}`
+  ngDoCheck(): void {
```

Comment:
terrible decision, check how many times this function calls just on "edit course" button click
![image](https://user-images.githubusercontent.com/50192515/184495919-4b38649c-2048-403a-8d5c-d5d7e239f37d.png)

Reply:
11 times
if you want wait for data and set plug data follow next way within template
![image](https://user-images.githubusercontent.com/50192515/184496019-af8a362a-a60d-48f5-83dc-eb208462540b.png)